src/IssueViz.Security.UnitTests/HotspotsList/HotspotsTableDataSourceTests.cs

```diff
@@ -79,6 +107,178 @@ public void Unsubscribe_RemovesEntriesFromSink()
+        {
+            var sink1 = new Mock<ITableDataSink>();
+            var sink2 = new Mock<ITableDataSink>();
+
+            var testSubject = CreateTestSubject();
+            testSubject.Subscribe(sink1.Object);
+            testSubject.Subscribe(sink2.Object);
+
+            var issueViz = CreateIssueViz();
+            testSubject.Add(issueViz);
+
+            const bool removeAllEntries = false;
+
+            sink1.Verify(x => x.AddEntries(It.Is((IReadOnlyList<ITableEntry> entries) =>
+                entries.Count == 1 &&
+                entries[0].Identity == issueViz
+            ), removeAllEntries), Times.Once);
+
+            sink2.Verify(x => x.AddEntries(It.Is((IReadOnlyList<ITableEntry> entries) =>
+                entries.Count == 1 &&
+                entries[0].Identity == issueViz
+            ), removeAllEntries), Times.Once);
+        }
+
+        [TestMethod]
+        public void Add_HasSubscribersToIssuesChangedEvent_SubscribersNotified()
+        {
+            var testSubject = CreateTestSubject();
+
+            IssuesChangedEventArgs suppliedArgs = null;
+            var eventCount = 0;
+            testSubject.IssuesChanged += (sender, args) => { suppliedArgs = args; eventCount++; };
+
+            var location1 = new Mock<IAnalysisIssueLocationVisualization>();
+            location1.SetupGet(x => x.CurrentFilePath).Returns("b.cpp");
+            var location2 = new Mock<IAnalysisIssueLocationVisualization>();
+            location2.SetupGet(x => x.CurrentFilePath).Returns("c.cpp");
+            var issueViz = CreateIssueViz("a.cpp", location1.Object, location2.Object);
+
+            testSubject.Add(issueViz);
+
+            eventCount.Should().Be(1);
+            suppliedArgs.Should().NotBeNull();
+            suppliedArgs.AnalyzedFiles.Should().BeEquivalentTo("a.cpp", "b.cpp", "c.cpp");
+        }
+
+        [TestMethod]
+        public void GetLocations_NoTableEntries_EmptyList()
+        {
+            var testSubject = CreateTestSubject();
+
+            var locations = testSubject.GetLocations("test.cpp");
+            locations.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetLocations_NoTableEntriesForGivenFilePath_EmptyList()
+        {
+            var testSubject = CreateTestSubject();
+            testSubject.Add(CreateIssueViz("file1.cpp"));
+
+            var locations = testSubject.GetLocations("file2.cpp");
+            locations.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetLocations_HasTableEntriesForGivenFilePath_ReturnsMatchingLocations()
+        {
+            var locationViz = new Mock<IAnalysisIssueLocationVisualization>();
+            locationViz.Setup(x => x.CurrentFilePath).Returns("somefile.cpp");
+
+            var issueViz1 = CreateIssueViz("somefile.cpp");
+            var issueViz2 = CreateIssueViz("someotherfile.cpp", locationViz.Object);
+            var issueViz3 = CreateIssueViz("somefile.cpp");
```

Comment:
You could change the casing of `issueViz3` to test that casing is handled correctly.